app/Traits/Likeable.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Traits;
+
+use App\Models\Like;
+
+trait Likeable
+{
+    public function like()
+    {
+        $like = new Like(['user_id' => auth()->id()]);
+        $this->likes()->save($like);
+    }
+
+    public function unLike()
+    {
+        $this->likes()->where('likeable_id', $this->id)->first()->delete();
```

Comment:
Why you use `first()` here?
Did `likes` table has unique constraint on `likeable_type` and `likeable_id`?